tests/mock_crust/utils.rs

```diff
@@ -295,20 +296,26 @@ pub fn poll_all(nodes: &mut [TestNode], clients: &mut [TestClient]) -> bool {
 /// Polls and processes all events, until there are no unacknowledged messages left and clearing
 /// the nodes' state triggers no new events anymore.
 pub fn poll_and_resend(nodes: &mut [TestNode], clients: &mut [TestClient]) {
+    // Expect advance at least three times, given NODE_CONNECT_TIMEOUT_SECS is 60s and
+    // ACK_TIMEOUT_SECS is 20s
+    let mut num_of_advance = 0;
     for _ in 0..MAX_POLL_CALLS {
         if poll_all(nodes, clients) {
             let mut call_count = 1;
-            while resend_unacknowledged(nodes, clients) && poll_all(nodes, clients) {
+            while poll_all(nodes, clients) {
                 call_count += 1;
                 assert_ne!(call_count,
                            MAX_POLL_CALLS,
-                           "Polling and resending unacknowledged has been called {} times.",
+                           "Polling and advance time has been called {} times.",
                            MAX_POLL_CALLS);
             }
-            nodes.iter_mut().foreach(|node| node.inner.clear_state());
-        } else {
+        } else if num_of_advance > 3 {
             return;
         }
+
+        FakeClock::advance_time(20 * 1000 + 1);
+        num_of_advance += 1;
+        let _ = poll_all(nodes, clients);
```

Comment:
Is that call necessary? `poll_all` will be called at the beginning of the next loop again anyway.